lib/menu.rb

```diff
@@ -0,0 +1,42 @@
+class Menu
+
+  attr_reader :basket, :list_of_items
+
+  def initialize
+    @list_of_items = { :naan => 1.00, :daal => 3.00, :rice => 2.00 }  # => {:naan=>1.0, :daal=>3.0, :rice=>2.0}
+    @basket = []                                                      # => []
+    @subtotal = []                                                    # => []
+  end
+
+  def order(item, quantity)
+    not_on_menu?(item,quantity)
+    @quantity = quantity
+    @price = list_of_items.fetch(item)
+    @list_of_items[item]
+    puts "Thank you for ordering #{quantity} x #{item}, costing £#{list_of_items.fetch(item)} each"  # => nil, nil
+    @basket << [item, list_of_items.fetch(item),quantity]                                            # => [[:naan, 1.0, 3]], [[:naan, 1.0, 3], [:rice, 2.0, 2]]
+    @subtotal << list_of_items.fetch(item)*quantity                                                  # => [3.0], [3.0, 4.0]
+  end
+
+  def basket_price
+    @basket_price = @subtotal.inject(:+).round(10)  # => 7.0
+  end
+
+  private
+  
+  def not_on_menu?(item,quantity)
+    raise "Item not on menu" unless @list_of_items.has_key?(item)
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`.<br>Style/SignalException: Use `fail` instead of `raise` to signal exceptions.